### ISSUE TITLE
Dark mode

### ISSUE THREAD

**Issue by fcoveram** [WordPress/openverse#3592]
## Description

Include a feature that changes the UI from light to a dark color scheme.

### Background

Dark mode is a product-wide setting that adapts the user interface to provide a comfortable experience in low-light environments and, in some cases, improve accessibility for those with visual impairments, as [pointed out by NN/g](https://www.nngroup.com/articles/dark-mode/).

As described in [Material Design](https://m2.material.io/design/color/dark-theme.html#usage) documentation, dark themes also reduce the luminance emitted by device screens and conserve battery power, as devices with OLED screens benefit from the ability to turn off black pixels at any time of day.

## Documents

- [ ] Project Proposal
- [ ] Implementation Plan(s)
- [ ] Design Proposal

## Milestones/Issues

TBD

**Comment by fcoveram** [comment]
## Update 2024-03-25

* Project proposal document is done, #3913 
* Project status changed to "In RFC"

**Comment by fcoveram** [comment]
# Update on 2024-05-01

* Frontend implementation plan is still in progress but very close to finish.
* Dark mode designs #3959 are done and the ticket was closed.
* #4155 was created to decide how to switch the site theme.
* #4157 was created to decide the feature introduction.
* #4232 was created requesting the addition of a new component for #4155
* In parallel, @fcoveram has been updating the Design Library with the new variables to have every Figma component in light and dark mode.

**Comment by zackkrida** [comment]
I am going to lead this project moving forward.

The [implementation plan](https://github.com/WordPress/openverse/issues/3958) for the project is approved and merged.

Next, I will create issues and milestones for the project. Once created I will outline the workflows and link to the milestones in the project issue body. At that point I will move the project to "On Hold" while we work on the Nuxt 3 migration. This will not be worked on until that project is completed, or is again put on hold due to limiting circumstances.

**Comment by zackkrida** [comment]
@fcoveram do you have an update yet on the following aspects of the designs?

- The finalized dark mode mockups
- The semantic color names to replace our existing colors ("yellow" to "complementary", for example)
- The mapping of dark mode colors to their light mode counterparts

**Comment by zackkrida** [comment]
## Update 05-10-2024

All issues and milestones have been created and added to the parent issue ✨. The project is ready to be worked on but will be moved to "On Hold".

There is some remaining work around color names that I raised [here](https://github.com/WordPress/openverse/issues/4268#issuecomment-2103597982) and am awaiting reply.

**Comment by fcoveram** [comment]
## Update 14-05-2024

* #4268 and #4325 were created to specify the new color system and the transition from current the current 

cc: @zackkrida

**Comment by zackkrida** [comment]
# Update 7-31-2024

I've initialized work on dark mode in light of the successful Nuxt 3 deployment. I've put up a PR for the first part of the "color palette" milestone. That milestone introduces the dark mode colors but doesn't expose them to end users.

https://github.com/WordPress/openverse/pull/4687

I've also created an initial PR to add the dark mode color palette. It's close to being ready for review:

https://github.com/WordPress/openverse/pull/4690

After that, the first milestone will be closed when we add visual regression tests and store screenshots for the dark version of components.

**Comment by zackkrida** [comment]
# Update 2024-08-12

The first dark mode milestone is nearly completed, with one issue remaining #4305.

But first, we need to do two things:

- review the dark mode for improvements, issues, or concerns. @fcoveram will be able to start this next week. 
- Complete the [second milestone](https://github.com/WordPress/openverse/milestone/32), which creates the UI and code necessary to toggle and introduce dark mode.

Once those are both completed, we can "lock in" the visual regression tests for dark mode. 

Given that our Storybook Playwright tests are paused, I will constrain the scope of that issue to creating dark mode _page_ screenshots for now.

**Comment by zackkrida** [comment]
## Update 2024-08-27

Progress! Dark mode is under extensive community review in #4783, with @obulat already pushing some improvements in #4808.

Work has begun on the [second milestone](https://github.com/WordPress/openverse/milestone/32). I've started writing the basic prerequisites in #4809 and #4810, while @dhruvkb plans to implement the Dark mode toggle component.

**Comment by fcoveram** [comment]
Yesterday, I spent half a day cleaning up some [views](https://www.figma.com/design/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?node-id=344-285&m=dev) to support dark mode. If you have edit permission in the Design Library, you can easily switch the theme of mockup sections marked as "ready for dev."

To avoid unintentional changes in the file, it is highly recommended that you switch between design and dev modes (`Shift + D`) and use the hand tool (`H`) to move around the file.

https://github.com/user-attachments/assets/52f0edcc-ca47-422b-a412-954325970713

**Comment by dhruvkb** [comment]
## Update 2024-09-19

- The new feature notice has been added in PR #4922, this closes out the dark mode feature introduction issue #4157.

> [!IMPORTANT]
> The notice currently points to the Make site. The URL must be updated when the Make post has been published.
> https://github.com/WordPress/openverse/blob/3ac64779288c09db9ba7ff677a265b3a0c4a3984/frontend/src/components/VFeatureNotice/VDarkModeFeatureNotice.vue#L16

**Comment by obulat** [comment]
## Update 2024-10-30

### Done

The work on the two dark mode milestones has finished. Here are some bigger issues closed recently:
- #4911 
- #4936 
- #5026
We have also finished adding all dark mode visual regression tests (#4305)

### To do
New issues were identified after the milestones were closed:
- Prevent layout switching when the theme switcher is rendered (#5085)
- Change the text color on text-heavy pages to `text-secondary` for easier reading

### Follow-up work

According to the [project launch plan](https://docs.openverse.org/projects/proposals/dark_mode/20240325-implementation_plan_dark_mode.html#launch-plan), we will need to write a make post and start the launch after the issues in the todo are done. Note that we might need to do one more testing after the dark mode is launched on staging before publicizing the launch.

**Comment by fcoveram** [comment]
I asked in the [amplification request ticket](https://github.com/WordPress/Marketing-Team/issues/403) for the next steps regarding communicating this project publicly and now waiting for a response.

**Comment by fcoveram** [comment]
# Update 2024-12-09

* Communication work for announcing the launch is in progress in [this document](https://docs.google.com/document/d/1lJ_a4d3dSj4HcMtixqHqFWkUk4e63eg_oVaZRL0CjWA/edit?usp=sharing) and soon to finish
